src/test/groovy/unit/org/pih/warehouse/order/OrderSpec.groovy

```diff
@@ -0,0 +1,107 @@
+package unit.org.pih.warehouse.order
+
+import grails.testing.gorm.DomainUnitTest
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderAdjustment
+import org.pih.warehouse.order.OrderItem
+import org.pih.warehouse.order.OrderItemStatusCode
+import spock.lang.Specification
+
+class OrderSpec extends Specification implements DomainUnitTest<Order> {
+    void "Order.getCanGenerateInvoice() should return true when there is an order adjustment not tied to any order item that hasn't already been invoiced"() {
+        given:
+        Order order = Spy(Order) {
+            getHasPrepaymentInvoice() >> hasPrepaymentInvoice
+        }
+        Set<OrderAdjustment> orderAdjustments = [
+                Mock(OrderAdjustment),
+        ]
+        order.setOrderAdjustments(orderAdjustments)
+
+        expect:
+        order.canGenerateInvoice == canGenerateInvoice
+
+        where:
+        hasPrepaymentInvoice || canGenerateInvoice
+        false                || false
+        true                 || false
```

Comment:
shouldn't this condition return true? You have an un-invoiced adjustment that has no associated order items

Reply:
that's interesting, why this test doesn't fail

Reply:
Oh, got the reason, I didn't take order status into account (orderAdjustment.isInvoiceable takes into account the fact if order is placed)